pyntcloud/io/stl.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+from stl import mesh
+import pandas as pd
+
+def read_stl(filename):
+    stl = mesh.Mesh.from_file(filename)
+    length1 = np.size(stl.x.flatten(),0)
+    #length2 = np.size(stl.normals.flatten(),0)
+#    length3 = np.size(stl.v0.flatten())
+#    
+    index1 = pd.RangeIndex(start=0,stop=length1)
+    #index3 = pd.RangeIndex(start=0,stop=length3)
+
+    x = stl.x.flatten()
+    y = stl.y.flatten()
+    z = stl.z.flatten()
+
+#    nx = stl.normals[:,0]
+#    ny = stl.normals[:,1]
+#    nz = stl.normals[:,2]
+
+#    v0 = stl.v0.flatten()
+#    v1 = stl.v1.flatten()
+#    v2 = stl.v2.flatten()
+
+#    x = x.reshape(length1,1)
+#    y = y.reshape(length1,1)
+#    z = z.reshape(length1,1)
+
+#    nx = nx.reshape(length2,1)
+#    ny = ny.reshape(length2,1)
+#    nz = nz.reshape(length2,1)
+
+#    v0 = v0.reshape(length3,1)
+#    v1 = v1.reshape(length3,1)
+#    v2 = v2.reshape(length3,1)
+
+    #stl_points = pd.DataFrame([1,1],0,columns=['x','y','z'])
+    data = {}
+#    data['points'] = pd.DataFrame(np.zeros([length1,6]),index=index1,columns=['x','y','z','nx','ny','nz'])
+    data['points'] = pd.DataFrame(np.zeros([length1,3]),index=index1,columns=['x','y','z'])
+
+    #normals = pd.DataFrame(np.zeros([length2,3]),index=index2,columns=['nx','ny','nz'])
+    
+#    data['mesh'] = pd.DataFrame(np.zeros([length3,3]),index=index3,columns=['v1','v2','v3'])
+    
+    data['points'].x = x
+    data['points'].y = y  
+    data['points'].z = z
+#    
+#    data['points'].nx = np.NaN
+#    data['points'].ny = np.NaN  
+#    data['points'].nz = np.NaN
+#    
+#    pd.DataFrame.update(data['points'],{'nx':nx,'ny':ny,'nz':nz})
+#        
+#    data['mesh'].v1 = v0
```

Comment:
This doesn't get the correct result either. I found this as the cleanest solution to extract the [face-vertex information](https://en.wikipedia.org/wiki/Polygon_mesh#Face-vertex_meshes) in the same format as the other 3D file formats:

```python 
face_vertex_indices = []
for face in mesh.vectors:
    vertices_indices = []
    for vertex in face:
        vertex_idx = np.flatnonzero((points == vertex).all(1))[0]
        vertices_indices.append(vertex_idx)
    face_vertex_indices.append(vertices_indices) 
    
data["mesh"] = pd.DataFrame(face_vertex_indices, columns=["v1", "v2", "v3"])
```